src/load/migrate-database.lisp

```diff
@@ -255,6 +255,29 @@
     (setf (catalog-distribution-rules catalog)
           (citus-distribute-schema catalog distribute))))
 
+(defun optimize-table-copy-ordering (catalog)
+  "Return a list of tables to copy over in optimized order"
+  (let ((table-list (copy-list (table-list catalog)))
+        (view-list  (copy-list (view-list catalog))))
+    ;; when materialized views are not supported, view-list is empty here
+    (cond
+      ((notevery #'null (mapcar #'table-row-count-estimate table-list))
+           (flet ((compare-row-count-estimates (a b)
+                    (declare (type (or null fixnum) a b))
+                    (> (or a 0) (or b 0))))
```

Comment:
Accessing uninitialized slots here. See the comment on `DEFSTRUCT TABLE`.

Reply:
Good point, now added a default value of zero and a type for this slot in the structure.